gcc/rust/typecheck/rust-tyty.h

```diff
@@ -1485,7 +1485,7 @@ class ClosureType : public BaseType, public SubstitutionRef
       parameter_types (std::move (parameter_types)),
       result_type (std::move (result_type)), id (id)
   {
-    LocalDefId local_def_id = id & DEF_ID_LOCAL_DEF_MASK;
+    LocalDefId local_def_id = id.localDefId;
     rust_assert (local_def_id != UNKNOWN_LOCAL_DEFID);
```

Comment:
This is more likely a separate PR for the future (as I suspect other constants could use similar help), but having this be a `LocalDefId::Unknown` class-level constant rather than a `#define` or `enum` would help with readability IMO.

Reply:
yep this is a good-first-pr i have been meaning to create